### PR TITLE
[#777] force reload on change dropdown selector event

### LINKED ISSUE #777: [Justice Counts] Uploaded Files bug
While debugging a separate issue, I noticed that when you are in the 'Uploaded Files' page within 'Settings, if you change agencies, the url changes but the page does not update:

https://images.zenhubusercontent.com/529290376/f2a6fecb-f256-4eee-9c29-07954e8ac761/loom_message___19_july_2023.mp4

Steps to reproduce:
1. Log in to Publisher
2. Select the following Agency: Columbus Police Department
3. Navigate to Setting > Uploaded Files (at the time of writing this ticket, CPD had attempted 1 file upload)
4. Select the following Agency: Iowa Department of Corrections (IDOC)
At this point, the URL and 'Welcome, <Name> at Iowa Department of Corrections (IDOC)' banner should have changed. However the contents of the Uploaded Files page has not.
5. Navigate to 'Team Management' and then back to 'Uploaded Files'
At this point, the page should have refreshed and populated the 'Uploaded Files' with the proper files associated with IDOC.

### PR DESCRIPTION
**Pull request by alfonsogoberjr**
## Description of the change

Force page reload when selecting Agency dropdown items

## Type of change

Bug

## Related issues

Closes #777

### Development

**This box MUST be checked by the submitter prior to merging**:
- [x] **Double- and triple-checked that there is no Personally Identifiable Information (PII) being mistakenly added in this pull request**

These boxes should be checked by the submitter prior to merging:
- [x] Tests have been written to cover the code changed/added as part of this pull request

### Code review

These boxes should be checked by reviewers prior to merging:

- [x] This pull request has a descriptive title and information useful to a reviewer
- [x] This pull request has been moved out of a Draft state, has no "Work In Progress" label, and has assigned reviewers
- [x] Potential security implications or infrastructural changes have been considered, if relevant

### CODE REVIEW

**Comment by lilidworkin** [comment]
Thanks for the quick fix @alfonsogoberjr ! @mxosman curious what you think of this approach? I imagine there was a reason we weren't doing this before? How are we currently updating the state on agency change, and is there a way to apply that existing logic to the Uploaded Files page?

**Comment by mxosman** [comment]
> Thanks for the quick fix @alfonsogoberjr ! @mxosman curious what you think of this approach? I imagine there was a reason we weren't doing this before? How are we currently updating the state on agency change, and is there a way to apply that existing logic to the Uploaded Files page?

Ooo - this is interesting. I probably would've just added the `agencyId` to the `UploadedFiles` dependency array so it re-renders:

https://github.com/Recidiviz/justice-counts/blob/7579815815a1e093ffc0a5c477388fb7e3d619b5/publisher/src/components/DataUpload/UploadedFiles.tsx#L313-L319)

But, this solution doesn't have that granular concern - it just updates the path (which has the new `agencyId`) and reloads the page. I'm actually not sure which would be best to do! On a component level, I think it makes sense to add the dependency because the useEffect should depend on the `agencyId` and refetch every time that changes - but I like how straightforward and agnostic this solution is.

File: yarn.lock [hunk 1]
```diff
@@ -4762,6 +4758,11 @@
     "@types/qs" "*"
     "@types/serve-static" "*"
 
+"@types/file-saver@^2.0.5":
```
Comment (mxosman) [inline yarn.lock:4761]: I think this yarn.lock might be an accidental commit?

**Comment by lilidworkin** [comment]
>  I probably would've just added the agencyId to the UploadedFiles dependency array so it re-renders:

Yea, I think this is probably the way to go, since that's how we handle agency switching everywhere else? What do you think @alfonsogoberjr ? Curious to hear if you considered that and what you think the pros/cons of each approach is.

**Comment by alfonsogoberjr** [comment]
> > Thanks for the quick fix @alfonsogoberjr ! @mxosman curious what you think of this approach? I imagine there was a reason we weren't doing this before? How are we currently updating the state on agency change, and is there a way to apply that existing logic to the Uploaded Files page?
> 
> Ooo - this is interesting. I probably would've just added the `agencyId` to the `UploadedFiles` dependency array so it re-renders:
> 
> https://github.com/Recidiviz/justice-counts/blob/7579815815a1e093ffc0a5c477388fb7e3d619b5/publisher/src/components/DataUpload/UploadedFiles.tsx#L313-L319
> 
> )
> But, this solution doesn't have that granular concern - it just updates the path (which has the new `agencyId`) and reloads the page. I'm actually not sure which would be best to do! On a component level, I think it makes sense to add the dependency because the useEffect should depend on the `agencyId` and refetch every time that changes - but I like how straightforward and agnostic this solution is.

Yeah way better

**Review by mxosman — APPROVED** [review]
LGTM!

**Pull request merged**